modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java

```diff
@@ -1918,7 +1927,19 @@ private static void doPatchReplaceOnPathWithoutFiltersForLevelOne(AbstractSCIMOb
             }
 
         } else {
-            //create and add the attribute
+            // Check whether the patched attributes are permissions of Roles.
+            if (schema.isSchemaAvailable(SCIMConstants.ROLE_SCHEMA_URI) &&
+                    SCIMConstants.RoleSchemaConstants.PERMISSIONS.equalsIgnoreCase(attributeParts[0])) {
```

Comment:
Can we guarantee the array index?

Reply:
Yes. This array is referenced before this instance as well. If the array is null (i.e. no path value has been specified in the patch request payload), then the execution follows a different path anyway.